src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -2438,6 +2438,29 @@ static inline __m128 log_ps(__m128 x)
     return x;
 }
 
+inline Rpp32f rpp_hsum_ps(__m128 x)
+{
+    __m128 shuf = _mm_movehdup_ps(x);        // broadcast elements 3,1 to 2,0
+    __m128 sums = _mm_add_ps(x, shuf);
```

Comment:
why aren't we using built-in hadd instructions?

Reply:
Hi Rajy,
With the usage of built-in  _mm_hadd_ps, the performance was slower,  so we used the above alternate method for doing sum of all values in __m128 register. The above approach is nearly 20% faster than the approach with built-in instruction

Reply:
Can you please make a sample code with both ways. We need to run it on different CPU versions

Reply:
H Rajy,
Adding a snippet of sample for both variations for your reference
![image](https://github.com/ROCm/rpp/assets/89194005/f6350bbf-f3c6-42df-945b-1cb5162ee88b)

In Intel intrinsic guide, _mm_hadd_ps latency is available only for Skylake 
On Skylake, Built-in variant takes a total latency of 19 cycles whereas custom variant takes 14 cycles